var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -256,16 +260,43 @@ def write_library_dirs(f, dirs):
             if '^netlib-lapack' in spec:
                 # netlib requires blas and lapack listed
                 # separately so that scipy can find them
+                extra_link_args = ''
+                rpath = ''
+                if '^netlib-lapack~shared' in spec:
+                    if "xlf" in env['SPACK_F77']:
+                        bin_index = env['SPACK_F77'].find("/bin")
```

Comment:
This could break if `/bin` is in the path twice. What about something like `os.path.dirname(os.path.dirname(self.compiler.f77))`?